bundles/org.openhab.binding.netatmo/src/main/java/org/openhab/binding/netatmo/internal/api/ApiResponse.java

```diff
@@ -0,0 +1,46 @@
+/**
+ * Copyright (c) 2010-2021 Contributors to the openHAB project
+ *
+ * See the NOTICE file(s) distributed with this work for additional
+ * information.
+ *
+ * This program and the accompanying materials are made available under the
+ * terms of the Eclipse Public License 2.0 which is available at
+ * http://www.eclipse.org/legal/epl-2.0
+ *
+ * SPDX-License-Identifier: EPL-2.0
+ */
+package org.openhab.binding.netatmo.internal.api;
+
+import org.eclipse.jdt.annotation.NonNullByDefault;
+
+/**
+ * The {@link ApiResponse} models a response returned by API call
+ *
+ * @author Gaël L'hopital - Initial contribution
+ */
+@NonNullByDefault
+public class ApiResponse<T> {
+    /**
+     * The {@link Ok} models a response that only holds a status
+     * toward the request sent to the API
+     */
+    static class Ok extends ApiResponse<String> {
+        private static final String SUCCESS = "ok";
+
+        public boolean failed() {
+            return !SUCCESS.equals(getStatus());
+        }
+    }
+
+    private String status = "";
+    private @NonNullByDefault({}) T body;
```

Comment:
Warning:
`ApiResponse.java:[37,37] The field body may not have been initialized, whereas its type 'T' is a free type variable that may represent a '@NonNull' type`

Reply:
I did not find any suitable way to address this.